pkg/querier/blocks_store_queryable.go

```diff
@@ -48,7 +55,7 @@ type BlocksStoreSet interface {
 
 	// GetClientsFor returns the store gateway clients that should be used to
 	// query the set of blocks in input.
-	GetClientsFor(blockIDs []ulid.ULID) (map[BlocksStoreClient][]ulid.ULID, error)
+	GetClientsFor(blockIDs []ulid.ULID, blacklist map[ulid.ULID][]string) (map[BlocksStoreClient][]ulid.ULID, error)
```

Comment:
Can we extend comment to explain usage of blacklist? (Comment from https://github.com/cortexproject/cortex/pull/2695/files#diff-d7623e8d715cf64e2bad62001eac549bR362 is fine)